packages/frontend/src/components/UserSettings/AllowedDomainsPanel/index.tsx

```diff
@@ -1,161 +1,186 @@
-            methods.setValue(
+        if (allowedEmailDomainsData) {
+            setFieldValue('emailDomains', allowedEmailDomainsData.emailDomains);
+            setFieldValue('role', allowedEmailDomainsData.role);
+            setFieldValue(
                 'projects',
-                projectOptions.filter(({ value }) =>
-                    data.projectUuids.includes(value),
-                ),
+                projectOptions
+                    .filter(({ value }) =>
+                        allowedEmailDomainsData.projectUuids.includes(value),
+                    )
+                    .map(({ value }) => value),
             );
         }
-    }, [data, methods, projectOptions]);
+    }, [allowedEmailDomainsData, projectOptions, setFieldValue]);
 
-    const handleUpdate = (values: FormData) => {
+    const handleOnSubmit = form.onSubmit((values) => {
         const role =
             values.emailDomains.length > 0
                 ? values.role
                 : OrganizationMemberRole.VIEWER;
         const projectUuids =
-            role === OrganizationMemberRole.MEMBER
-                ? values.projects.map(({ value }) => value)
-                : [];
+            role === OrganizationMemberRole.MEMBER ? values.projects : [];
         mutate({
             emailDomains: values.emailDomains,
             role,
             projectUuids,
         });
-    };
+    });
 
-    return (
-        <FormWrapper>
-            <Form
-                name="login"
-                methods={methods}
-                onSubmit={handleUpdate}
-                disableSubmitOnEnter
-                isLoading={isLoading}
-            >
-                <TagInput
-                    label="Allowed email domains"
+    return isSuccess ? (
+        <form name="allowedEmailDomains" onSubmit={handleOnSubmit}>
+            <Stack>
+                <MultiSelect
                     name="emailDomains"
+                    label="Allowed email domains"
                     placeholder="E.g. lightdash.com"
                     disabled={isLoading}
-                    defaultValue={data?.emailDomains || []}
-                    rules={{
-                        validate: {
-                            isValidEmailDomain:
-                                isValidEmailDomain('Email domains'),
-                            isValidOrganizationDomain:
-                                isValidOrganizationDomain('Email domains'),
-                        },
+                    data={form.values.emailDomains.map((emailDomain) => ({
+                        value: emailDomain,
+                        label: emailDomain,
+                    }))}
+                    searchable
+                    creatable
+                    getCreateLabel={(query: string) => `+ Add ${query} domain`}
+                    defaultValue={form.values.emailDomains}
+                    onCreate={(value) => {
+                        if (!isValidEmailDomain(value)) {
```

Comment:
Should we check if there is at least one `.` in the string? 

wrong value
![Captura de ecrã 2023-05-17, às 16 28 18](https://github.com/lightdash/lightdash/assets/9117144/4405f037-f4cb-4765-8197-5d1ce7bd8d72)

Reply:
https://github.com/lightdash/lightdash/assets/7611706/293c2744-22e3-4b00-b4ed-53dd1034a5c3

here it is!